### PR TITLE
Clinical case user feedback

### LINKED ISSUE #388: [NIRC] Clinical Cases - Misc  updates
- [x] Hide 'Close Remarks' for non-vets in Cases form
- [x] Propagate 'Ordered By' from Treatment Orders to Treatments Given
- [x] SOAP Notes, make them not required when 'close date' is added
- [x] End treatment Ordered By
- [x] Duplicate Obs Orders
- [x] Ordered by in views
- [x] Code Review @labkey-bpatel 
- [x] Code Review Feedback
- [x] TC and Merge

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
Handle a variety of clinical case updates from user feedback and testing.

#### Related Pull Requests
* https://github.com/LabKey/ehrModules/pull/831

#### Changes
* Hide 'Close Remarks' for non-vets in Cases form
* Propagate 'Ordered By' from Treatment Orders to Treatments Given
* SOAP Notes, make them not required when 'close date' is added
* End treatment Ordered By
* Duplicate Obs Orders
* Ordered by in views
* Observation schedule status updates

### CODE REVIEW

File: nirc_ehr/resources/web/nirc_ehr/data/CaseClientStore.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+Ext4.define('NIRC_EHR.data.CaseClientStore', {
+    extend: 'EHR.data.DataEntryClientStore',
+
+    getExtraContext: function(){
+        // Submit all of the weights in this batch as a separate property on the extra context,
+        // so that server-side validation can see them all at once for validation purposes (and not just rely
+        // on the most recent weight entry that's already been saved to the database)
```
Comment (labkey-bpatel) [inline nirc_ehr/resources/web/nirc_ehr/data/CaseClientStore.js]: Is this comment relevant here?

File: nirc_ehr/resources/web/nirc_ehr/window/MarkTreatmentCompletedWindow.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2013-2019 LabKey Corporation
```
Comment (labkey-bpatel) [inline nirc_ehr/resources/web/nirc_ehr/window/MarkTreatmentCompletedWindow.js:2]: 2024-2025

**Review by labkey-bpatel — APPROVED** [review]
Overall, it looks great, these are the few things I noticed:
- On Cases form, the Id is not being passed to Observation Orders and Treatment Orders (sounds like this is wip)
- Under 'Active Medication Orders', the view name is 'Clinical Active Treatment Orders' - should we change 'Treatment' to 'Medication' for consistency?
- Under 'Active Observation Orders', isActive filter appears to be missing.

File: nirc_ehr/resources/web/nirc_ehr/data/CaseClientStore.js (hunk 1 above)
Comment (labkey-martyp) [reply nirc_ehr/resources/web/nirc_ehr/data/CaseClientStore.js]: updated

**Pull request merged**